src/utils/parallelPaginatedRequest.js

```diff
@@ -0,0 +1,75 @@
+import axios from 'axios';
+
+async function fetchPage(apiUrl, requestConfig, page, pageSize, maxRetries) {
+  let attempt = 0;
+  let data = null;
+
+  while (attempt < maxRetries) {
+    attempt++;
+
+    try {
+      const response = await axios.get(`${apiUrl}&page=${page}&page_size=${pageSize}`, requestConfig);
+
+      if (response.status !== 200) {
+        throw new Error(`HTTP error ${response.status}`);
+      }
+
+      data = response.data.data;
+      break;
+    } catch (error) {
+      console.error(`Error fetching page ${page} (attempt ${attempt}):`, error.message);
+    }
+  }
+
+  if (data === null) { // just logging in error, this place can be used to trigger a callback on maxRetries exhausted per page
+    console.error(`Failed to fetch page ${page} after ${maxRetries} attempts.`);
+  }
+
+  return data;
+}
+
+async function parallelPaginatedRequest(apiUrl, requestConfig = {}, { itemsPerPage = 150, maxRetries = 3, concurrencyLimit = 5, onPageFetch = null } = {}) {
```

Comment:
I'm ok having defaults I guess, but I see you added the defaults of the specific scenario you are working on (fetching map events). But this method is designed to be reused so maybe these are not the best generic defaults. I guess we shouldn't really have a default `itemsPerPage` since that will vary every time. The rest seems ok for me